internal/hooks/scan.go

```
@@ -0,0 +1,104 @@
+	Command string `json:"command"`
+}
+
+// Scan reads each settings file path in order and returns the combined hook
+// inventory. Missing files are silently skipped; malformed files are recorded
+// in Inventory.ParseErrors and do not abort the scan. The only error Scan
+// returns is for an unreadable (but existing) file's non-IsNotExist failure —
+// callers can otherwise treat the inventory as best-effort.
+func Scan(paths []string) (*Inventory, error) {
+	inv := &Inventory{}
+	for _, p := range paths {
+		data, err := os.ReadFile(p)
```

golangci-lint (2.12.2)

[medium] G304: Potential file inclusion via variable

(gosec)